internal/pkg/apply/apply.go

```diff
@@ -39,13 +42,42 @@ type Result struct {
 // Do executes the apply
 func (a *Apply) Do() (Result, error) {
 	fmt.Fprintf(a.Out, "Doing `cli-experimental apply`\n")
-	pods, err := a.Clientset.CoreV1().Pods("default").List(metav1.ListOptions{})
+	cl, err := client.New(config.GetConfigOrDie(), client.Options{})
 	if err != nil {
-		return Result{}, err
+		fmt.Fprintf(a.Out, "failed to create a client\n")
 	}
-	for _, p := range pods.Items {
-		fmt.Fprintf(a.Out, "Pod %s\n", p.Name)
+	for _, resource := range a.Resources {
+		u, ok := resource.(*unstructured.Unstructured)
+		if !ok {
+			fmt.Fprintf(a.Out, "Not able to convert runtime object to unstructured\n")
```

Comment:
Maybe we should just change the type of Resources to be a list of Unstructured pointers.